mm/shm/shm_initialize.c

```diff
@@ -25,6 +25,7 @@
 #include <nuttx/config.h>
 
 #include <assert.h>
+#include <debug.h>
```

Comment:
Why `#include <debug.h>` is needed here?

Reply:
You get a link time error due to missing symbol for each call to shmerr(..); Which expands to merr(...) -> _err(...)

So this fixes the dependency list

Reply:
Interestingly I was getting an error for `minfo()` even with `#include <debug.h>`. I was trying to build sama5d4-ek:knsh with `MM_SHM` enabled and was having problems. Not sure why I had problems but I ran out of time for today because of other work.

Reply:
This:
https://github.com/apache/incubator-nuttx/blob/master/arch/arm/src/armv7-a/arm_addrenv_shm.c

File seems to be missing #include <debug.h> as well, maybe adding it there will fix your issue?

I tested compiling for riscv target only, which obviously works.